lib/qa/authorities/discogs/discogs_instance_builder.rb

```diff
@@ -0,0 +1,127 @@
+        # and release information. Process the "name" field first, then the "descriptions"
+        # In unusual cases, there can be multiple playing speeds. Need to distinguish among them.
+        count = 1
+        return stmts unless response["formats"].present?
+        response["formats"].each do |format|
+          stmts.concat(build_format_name_stmts(format["name"])) # if format["name"].present?
+          # Now process playing speed, playback channel and release info
+          stmts.concat(build_format_desc_stmts(format["descriptions"], count)) # if format["descriptions"].present?
+          count += 1
+        end
+        stmts # w/out this line, building the graph throws an undefined method `graph_name=' error
+      end
+
+      def get_labels_stmts(response)
+        stmts = []
+        # Various roles are defined as provision activities, such as the record company (or label),
+        # the publishing house, the recording studio, etc. These are defined separately in the Discogs
+        # data, so combine them into a single array to be processed in one iteration.
+        provision_activity_array = response["labels"] if response["labels"].present?
+        provision_activity_array += response["companies"] if response["companies"].present?
+        stmts += build_provision_activity_stmts(provision_activity_array) if provision_activity_array.present?
+        stmts # w/out this line, building the graph throws an undefined method `graph_name=' error
+      end
+
+      def get_identifiers_stmts(response)
+        stmts = []
+        # The Discogs data includes identifiers such as side label codes and rights society codes.
+        count = 1
+        return stmts unless response["identifiers"].present?
+        response["identifiers"].each do |activity|
+          stmts << contruct_stmt_uri_object("Instance1", "http://id.loc.gov/ontologies/bibframe/identifiedBy", "Identifier#{count}")
+          stmts << contruct_stmt_uri_object("Identifier#{count}", rdf_type_predicate, "http://id.loc.gov/ontologies/bibframe/Identifier")
+          stmts << contruct_stmt_literal_object("Identifier#{count}", rdfs_label_predicate, activity["value"])
+          count += 1
+        end
+        stmts # w/out this line, building the graph throws an undefined method `graph_name=' error
+      end
+
+      def build_format_desc_stmts(descs, count)
+        stmts = []
+        return stmts unless descs.present?
+        descs.each do |desc|
+          # map discogs description field to the corresponding LOC type
+          df = discogs_formats[desc.gsub(/\s+/, "")]
+          if df.present?
+            stmts += build_format_characteristics(df, count)
+          else
+            stmts << contruct_stmt_literal_object("Instance1", "http://id.loc.gov/ontologies/bibframe/editionStatement", desc)
+          end
+        end
+        stmts # w/out this line, building the graph throws an undefined method `graph_name=' error
```

Comment:
If descs is an array, descs.each will return the descs array.  So it makes sense to me that you would need to return stmts.

Reply:
removed the closing line